heft-plugins/heft-jest-plugin/src/TerminalWritableStream.ts

```diff
@@ -0,0 +1,28 @@
+import { ITerminal } from '@rushstack/node-core-library';
+import { Writable } from 'stream';
+
+// Regex to filter out screen clearing directives
+// eslint-disable-next-line no-control-regex
+const FILTER_REGEX: RegExp = /\x1B\[2J\x1B\[0f|\x1B\[2J\x1B\[3J\x1B\[H/g;
+
+export class TerminalWritableStream extends Writable {
+  private readonly _terminal: ITerminal;
+
+  public constructor(terminal: ITerminal) {
+    super({
+      objectMode: false,
+      decodeStrings: false,
+      defaultEncoding: 'utf-8'
+    });
+
+    this._terminal = terminal;
+  }
+
+  // eslint-disable-next-line @typescript-eslint/no-explicit-any
+  public _write(chunk: any, encoding: string, callback: (error?: Error | undefined) => void): void {
+    const stringified: string = typeof chunk === 'string' ? chunk : chunk.toString(encoding);
+    const filtered: string = stringified.replace(FILTER_REGEX, '');
```

Comment:
Can use the Ansi utility from node-core-library

Reply:
Only if I want to also strip all color coding and formatting, which I don't. This is only used to strip the "clear console" directives.

Reply:
Might be nice then to break that part out into the Ansi tool as well then so it can be re-used elsewhere.